src/sentry/api/endpoints/organization_details.py

```diff
@@ -655,6 +670,49 @@ def update_tracked_data(model):
         model.__data = UNSAVED
 
 
+class DeleteConfirmationArgs(TypedDict):
+    username: str
+    ip_address: str
+    deletion_datetime: datetime
+    organization: Organization
+    countdown: int
+
+
+def send_delete_confirmation(delete_confirmation_args: DeleteConfirmationArgs):
+    from sentry import options
+    from sentry.utils.email import MessageBuilder
+
+    organization = delete_confirmation_args.get("organization")
+    username = delete_confirmation_args.get("username")
+    user_ip_address = delete_confirmation_args.get("ip_address")
+    deletion_datetime = delete_confirmation_args.get("deletion_datetime")
+    countdown = delete_confirmation_args.get("countdown")
+
+    url = organization.absolute_url(
+        reverse("sentry-restore-organization", args=[organization.slug])
+    )
```

Comment:
It didn't entirely make sense to me that this lived within the organization model, so I pulled it up into the controller instead.

Reply:
:+1: stateless module methods are always nice to see.